lib/rules/no-useless-assignment.js

```diff
@@ -229,6 +230,22 @@ module.exports = {
              */
             function verifyAssignmentIsUsed(targetAssignment, allAssignments) {
 
+                // Check if the variable is used in a catch clause
+                const isUsedInCatchClause = targetAssignment.variable.references.some(reference => {
+                    const id = reference.identifier;
+
+                    if (reference.isRead()) {
+                        return [...scopeStack.catchClauses].some(catchClause =>
```

Comment:
```suggestion
                        return [...target.catchClauses].some(catchClause =>
```

Since this function checks the given `target` ScopeStack.